move/axelar_gateway/sources/types/proof.move

```diff
@@ -60,6 +61,33 @@ public(package) fun recover_pub_key(
     ecdsa::secp256k1_ecrecover(&self.bytes, message, 0)
 }
 
+/// Validates the signatures of a message against the signers.
+/// The total weight of the signatures must be greater than or equal to the threshold.
+/// Otherwise, the error `ELowSignaturesWeight` is raised.
+public(package) fun validate(
+    self: &Proof,
+    message: vector<u8>
+) {
+    let signers = self.signers();
+    let signatures = self.signatures();
+    assert!(signatures.length() != 0, ELowSignaturesWeight);
+
+    let threshold = signers.threshold();
+    let mut total_weight: u128 = 0;
+
+    signatures.do_ref!<Signature>(|signature| {
+        let pub_key = signature.recover_pub_key(&message);
+
+        let weight = signers.find_signer_weight(&pub_key);
```

Comment:
This makes it asymptotically worse. Find would need to start from the next index but Move might not allow doing that cleanly.

Reply:
`fold` could be used but I don't see it being readable. I prefer sticking with the while loop